include/ginkgo/core/matrix/csr.hpp

```diff
@@ -549,7 +598,13 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
     void convert_to(Csr<ValueType, IndexType> *result) const override
     {
         bool same_executor = this->get_executor() == result->get_executor();
-        EnableLinOp<Csr>::convert_to(result);
+        // NOTE: as soon as strategies are improved, this can be reverted
```

Comment:
unrelated to your changes: Is there a reason why this function (and move_to(...)) is implemented in the header instead of csr.cpp?

Reply:
I'm unsure, usually we do, for example for `Coo`:
```c++
    using EnableLinOp<Coo>::convert_to;
    using EnableLinOp<Coo>::move_to;
```
So they are defined in the header in that case, maybe that was done intentionally to keep that situation when we fixed some strategy issues?